File: 3239-minimum-number-of-flips-to-make-binary-grid-palindromic-i/3239-minimum-number-of-flips-to-make-binary-grid-palindromic-i.cpp
class Solution {
public:
    int minFlips(vector<vector<int>>& grid) {
        int row = grid.size(), col = grid[0].size();
        int ans1 = 0, ans2 = 0;

        for (int i = 0; i < row; ++i) {
            for (int j = 0; j < col / 2; ++j) {
                if (grid[i][j] != grid[i][col - 1 - j]) {
                    ans1++;
                }
            }
        }
        for (int j = 0; j < col; ++j) {
            for (int i = 0; i < row / 2; ++i) {
                if (grid[i][j] != grid[row - 1 - i][j]) {
                    ans2++;
                }
            }
        }
    
        return min(ans1, ans2);
    }
};

File: LeetCode/3239-minimum-number-of-flips-to-make-binary-grid-palindromic-i/3239-minimum-number-of-flips-to-make-binary-grid-palindromic-i.py
class Solution:
    def minFlips(self, grid: List[List[int]]) -> int:
        row = len(grid)
        col = len(grid[0])
        ans1 = 0
        ans2 = 0

        for i in range(row):
            for j in range(col // 2):
                if grid[i][j] != grid[i][col - 1 - j]:
                    ans1 += 1

        for j in range(col):
            for i in range(row // 2):
                if grid[i][j] != grid[row - 1 - i][j]:
                    ans2 += 1
    
        return min(ans1, ans2)